packages/nuxt/src/app/composables/router.ts

```diff
@@ -17,6 +17,9 @@ export const useRouter = () => {
 }
 
 export const useRoute = (): RouteLocationNormalizedLoaded => {
+  if (process.dev && isProcessingMiddleware()) {
+    console.warn('[nuxt] Calling `useRoute` within middleware may lead to misleading results. Instead, use the (to, from) arguments passed to the middleware to access the new and old routes.')
```

Comment:
Should we continue on getting results? (not throwing an error, might be harder to make developers see/condider the warning)

Reply:
Tricky. I am assuming that there might _possibly_ be a reason someone might want to use it, but happy to make a fatal error if you think that is a better DX.

Reply:
Lets do it in next releases then

Reply:
@pi0 @danielroe let's imagine we have composable which can be used in middleware or components. 

```js
export const useApi = (
    path?: string,
    options?: FetchOptions<'json'>
  ) => {
    let defaultPath = path
  
    if (!defaultPath) {
      defaultPath = useRoute().fullPath
    }
  })
  ```
  
In the middleware I can provide path from `to` object 
```js
useApi(to.fullPath)
```
But in the vue components I don't have to. 

But with this implementation we won't be able to use it in this way and we always have to pass path to composable which could be use thanks to `useRoute`. Why ?

Reply:
Unfortunately we got warnings in our library even If I add this condition

Reply:
Then it's not related, or you are still using it somewhere it shouldn't be used. Would you create a new issue with a reproduction?

Reply:
sorry guys, I had some issues on my side, everything is ok.